packages/portals/backend/sirius-components-collaborative-portals/src/main/java/org/eclipse/sirius/components/collaborative/portals/handlers/AddPortalViewEventHandler.java

```diff
@@ -81,7 +82,7 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
                     if (optionalNewPortal.isPresent()) {
                         context.setNextPortal(optionalNewPortal.get());
                         payload = new SuccessPayload(addPortalViewInput.id(), List.of());
-                        changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, context.getEditingContext().getId(), context.getInput());
+                        changeDescription = new ChangeDescription(PortalChangeKind.PORTAL_VIEW_ADDITION.name(), context.getEditingContext().getId(), context.getInput());
```

Comment:
The sourceId of the change description should be your representation id, those changes are not made directly on the editing context. This can create some issues with the refresh (since we can't detect that such change has been made by your representation)

Reply:
This by itself could be enough to explain the bug you had. If you look at `EditingContextEventProcessor#setupChangeDescriptionSinkConsumer`, we look at the sourceId to find the representation to refresh first and then we refresh all the others. If you don't have the proper sourceId, you will never be refreshed before the representations view it will be based on the order of the subscription I think which is not stable over time.

Reply:
I fixed the sourceId, but without the 3rd commit I still have the issue.

Reply:
The sourceId was not the (only) issue, but the off-by-one refresh issue should be fixed in the new version.